File: 349.intersection-of-two-arrays.js
/*
 * @lc app=leetcode id=349 lang=javascript
 *
 * [349] Intersection of Two Arrays
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  const res = [];
  const lookup = new Set();
  for (const num of nums1) {
    lookup.add(num);
  }
  for (const num of nums2) {
    if (lookup.has(num)) {
      res.push(num);
      lookup.delete(num);
    }
  }
  return res;
};
// @lc code=end

File: 349.intersection-of-two-arrays.go
/*
 * @lc app=leetcode id=349 lang=golang
 *
 * [349] Intersection of Two Arrays
 */

package leetcode

// @lc code=start
func intersection(nums1 []int, nums2 []int) []int {
	lookup := map[int]bool{}
	res := []int{}
	for i := 0; i < len(nums1); i++ {
		lookup[nums1[i]] = true
	}
	for i := 0; i < len(nums2); i++ {
		if lookup[nums2[i]] {
			lookup[nums2[i]] = false
			res = append(res, nums2[i])
		}
	}
	return res
}

// @lc code=end
